### PR TITLE
#193, #324, #327, #717, #851 Replaced the splash screen with a preload

### LINKED ISSUE #193: IDE does not load until you return to the Arduino IDE screen and click on something
If you attempt to multitask while the IDE is loading, the IDE does not load until you return to the Arduino IDE screen and click on something.

IDE Version

> Version: 2.0.0-beta.3
Date: 2021-02-26T10:48:13.025Z
CLI Version: 0.16.1 alpha [76f55490]

### LINKED ISSUE #324: Splash Screen Should not be Always on Top
**Describe the bug**
When launching the application, the splash screen will force itself on top of all other applications stop interaction with other applications.

**To Reproduce**

1. Start application
2. Wait for splash screen
3. Attempt to make another window active

**Expected behavior**
A user would expect to be able to use their computer without having to wait for the application to start, as this may take some time (especially on slower devices). The splash screen should continue to show on top once the application is launched but should not continue stay on top of applications made active after the splash screen is shown.

Always on top

**Desktop (please complete the following information):**
 - OS: Windows 10 Education & macOS Big Sur 11.2.3
 - Version: 2.0.0-beta.5-nightly.20210418

### LINKED ISSUE #327: Application start up is paused when window is minimised
**Describe the bug**
If the loading project Windows is minimised while the application is starting the starting process will not continue and the splash screen will stay visible indefinitely (or at least for an abnormally longer amount of time).

**To Reproduce**
1. Start application
2. Minimise the loading project window
3. Loading will not continue and splash screen stays visible 

**Expected behavior**
Starting will continue or the splash screen will be minimised along with the window

**Screenshots**
Long
(Stalls at splash screen for some time)

**Desktop (please complete the following information):**
 - OS: macOS Big Sur 11.2.3
 - Version: 2.0.0-beta.5-nightly.20210418

### PR DESCRIPTION
**Pull request by kittaakos**
### Motivation

Removes the splash screen and replaces the loading animation with a Theia preload. This PR does not contain the final design, in fact, I just copied the first Arduino SVG I found on the Internet and prepared the stub of the change.

### Change description

 - Removed the splash screen (status assets, services, etc.),
 - Added an HTML preload example.

### In Action:

`light` IDE2, `light` OS theme:

https://user-images.githubusercontent.com/1405703/182808523-924a3897-fd6b-4371-a6a7-6dcf67a1f46c.mp4

`light` IDE2, `dark` OS theme:

https://user-images.githubusercontent.com/1405703/182808615-ad484318-3ed5-48c4-899f-cce4ae78ac62.mp4

`dark` IDE2, `light` OS theme:

https://user-images.githubusercontent.com/1405703/182808769-4df69f39-709c-4c0e-a23b-201c589e33ac.mp4

`dark` IDE2, `dark` OS theme:

https://user-images.githubusercontent.com/1405703/182808921-c83b12e9-c3b9-4edf-9940-921de6bee27e.mp4

### Other information

Noteworthy things:
 - ~This PR depends on #1282 (I can make it independent if required)~ Done ✅ ,
 - When loading the preload, the IDE2 app, the themes, and the user-selected theme are unavailable, but the preload must have a background color. This example uses the hard-coded color values of the `editor.background` and `editor.foreground` theme constants. If the OS theme is `dark`, the `dark` editor colors will be used for the preload. Otherwise, the `light` one.

Closes #193
Closes #324
Closes #327
Closes #717
Closes #851

### Reviewer checklist

* [ ] PR addresses a single concern.
* [ ] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-ide/pulls) before creating one)
* [ ] PR title and description are properly filled.
* [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: electron-app/resources/preload.html [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                    <path
+                        d="M354.37 662.051h152.625v49.181H354.37zM1016.484 662.051h-51.671v-51.747h-49.348v51.747h-51.648v49.181h51.648v51.737h49.348v-51.737h51.671z" />
+                </g>
+            </svg>
+        </div>
+    </div>
+</body>
+
+</html>
```
Comment (kittaakos) [inline electron-app/resources/preload.html:121]: Missing newline.

File: electron-app/patch/frontend/index.js [hunk 2]
```diff
@@ -92,7 +92,7 @@ if (!window[ThemeServiceSymbol]) {
     arduinoLightTheme
   );
   themeService.startupTheme();
-  themeService.setCurrentTheme(defaultTheme);
+  // themeService.setCurrentTheme(defaultTheme); // TODO: check if window.localStorage.getItem('theme') === undefined, then set
```
Comment (kittaakos) [inline electron-app/patch/frontend/index.js:95]: Verify this

File: electron-app/package.json [hunk 3]
```diff
@@ -61,7 +61,7 @@
     },
     "generator": {
       "config": {
-        "preloadTemplate": "<div class='theia-preload' style='background-color: rgb(237, 241, 242);'></div>"
+        "preloadTemplate": "./resources/preload.html"
```
Comment (kittaakos) [inline electron-app/package.json:64]: I doubt this is currently working in a bundled app.

File: electron-app/resources/preload.html (hunk 1 above)
Comment (kittaakos) [reply electron-app/resources/preload.html]: Done.

File: electron-app/patch/frontend/index.js (hunk 2 above)
Comment (kittaakos) [reply electron-app/patch/frontend/index.js]: Done,

File: electron-app/package.json (hunk 3 above)
Comment (kittaakos) [reply electron-app/package.json:64]: Done.

**Review by per1234 — APPROVED** [review]
I tested this on Windows and Linux and it fixes https://github.com/arduino/arduino-ide/issues/193, https://github.com/arduino/arduino-ide/issues/324, https://github.com/arduino/arduino-ide/issues/325, https://github.com/arduino/arduino-ide/issues/327 for me

Thanks so much Akos!

**Review by AlbyIanna — APPROVED** [review]
I tested it on MacOS and it works as expected 👍 
LGTM ✅

**Comment by kittaakos** [comment]
I had an offline chat with @91volt, and we agreed that I enhance this PoC to enable [`lottie`](https://github.com/airbnb/lottie) for the preload in case we want to support a more sophisticated solution than an SVG.

**Comment by kittaakos** [comment]
> I had an offline chat with @91volt, and we agreed that I enhance this PoC to enable [`lottie`](https://github.com/airbnb/lottie) for the preload in case we want to support a more sophisticated solution than an SVG.

I have pushed the updated version. Please review.

**Review by per1234 — APPROVED** [review]
Updated version looks great. Excellent work Ernesto and Akos!

**Comment by kittaakos** [comment]
As requested offline, I switched to a preload GIF. Please review.

File: electron-app/preload/index.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// TODO: use `lottie` to load the animation from JSON.
+// Add `lottie-web` as a `devDependency` and here do `require('lottie-web').loadAnimation`
+// Currently IDE2 uses a GIF to render the preload.
```
Comment (kittaakos) [inline electron-app/preload/index.js:3]: add newlinw

**Comment by 91volt** [comment]
With the gif In term of smoothness of the animation is much better, thank you @kittaakos.
The only problem I see is the stretched aspect ratio of the gif:
Schermata 2022-08-08 alle 14 51 32

**Comment by ubidefeo** [comment]
@91volt 
I have the same stretch issue
CleanShot 2022-08-08 at 18 59 49@2x

**Review by ubidefeo — APPROVED** [review]
Works as expected
thank you

**Comment by ckuethe** [comment]
This is lovely. Nice fast startup, and the little bouncy animation make it feel like the startup is doing something.

**Pull request merged**